api/v1alpha3/virtualmachine_conversion.go

```diff
@@ -12,6 +12,72 @@ import (
 	vmopv1 "github.com/vmware-tanzu/vm-operator/api/v1alpha4"
 )
 
+func Convert_v1alpha3_VirtualMachineStorageStatus_To_v1alpha4_VirtualMachineStorageStatus(
+	in *VirtualMachineStorageStatus, out *vmopv1.VirtualMachineStorageStatus, s apiconversion.Scope) error {
+
+	if err := autoConvert_v1alpha3_VirtualMachineStorageStatus_To_v1alpha4_VirtualMachineStorageStatus(in, out, s); err != nil {
+		return err
+	}
+
+	if in.Usage != nil {
+		out.Total = in.Usage.Total
+		if in.Usage.Other != nil {
+			if out.Used == nil {
+				out.Used = &vmopv1.VirtualMachineStorageStatusUsed{}
+			}
+			out.Used.Other = in.Usage.Other
+		}
+		if in.Usage.Disks != nil {
+			if out.Requested == nil {
+				out.Requested = &vmopv1.VirtualMachineStorageStatusRequested{}
+			}
+			out.Requested.Disks = in.Usage.Disks
+		}
+	}
+
+	return nil
+}
+
+func Convert_v1alpha4_VirtualMachineStorageStatus_To_v1alpha3_VirtualMachineStorageStatus(
+	in *vmopv1.VirtualMachineStorageStatus, out *VirtualMachineStorageStatus, s apiconversion.Scope) error {
+
+	if err := autoConvert_v1alpha4_VirtualMachineStorageStatus_To_v1alpha3_VirtualMachineStorageStatus(in, out, s); err != nil {
+		return err
+	}
+
+	if in.Total != nil {
+		if out.Usage == nil {
+			out.Usage = &VirtualMachineStorageStatusUsage{}
+			out.Usage.Total = in.Total
```

Comment:
Is this really only suppose to be set when `out.Usage` is nil, instead of the usual pattern of

```
if out.Usage == nil {
  out.Usage = &VirtualMachineStorageStatusUsage{}
}
out.Usage.Total = in.Total
```

The later ones that just check for `if out.Usage == nil ` won't set their fields